File: Programming EF Code First/chapter4/Chapter 4 VB/Fluent/DataAccess/LodgingConfiguration.vb
Imports System.Data.Entity.ModelConfiguration
Imports Model

Namespace DataAccess
  Public Class LodgingConfiguration
	  Inherits EntityTypeConfiguration(Of Lodging)
	Public Sub New()
      Me.Property(Function(l) l.Name).IsRequired().HasMaxLength(200)

      Me.Property(Function(l) l.Owner).IsUnicode(False)

      Me.Property(Function(l) l.MilesFromNearestAirport).HasPrecision(8, 1)

      HasOptional(Function(l) l.PrimaryContact).WithMany(Function(p) p.PrimaryContactFor)

	  HasOptional(Function(l) l.SecondaryContact).WithMany(Function(p) p.SecondaryContactFor)
	End Sub
  End Class
End Namespace

File: Programming EF Code First/chapter4/Chapter4/BAGA_FluentAPI/DataAccess/LodgingConfiguration.cs
using System.Data.Entity.ModelConfiguration;
using Model;

namespace DataAccess
{
  public class LodgingConfiguration : EntityTypeConfiguration<Lodging>
  {
    public LodgingConfiguration()
    {
      Property(l => l.Name)
        .IsRequired()
        .HasMaxLength(200);

      Property(l => l.Owner)
        .IsUnicode(false);

      Property(l => l.MilesFromNearestAirport)
        .HasPrecision(8, 1);

      HasOptional(l => l.PrimaryContact)
        .WithMany(p => p.PrimaryContactFor);

      HasOptional(l => l.SecondaryContact)
        .WithMany(p => p.SecondaryContactFor);
    }
  }
}